consensus/polybft/fsm.go

```diff
@@ -192,25 +190,34 @@ func (f *fsm) BuildProposal(currentRound uint64) ([]byte, error) {
 	return stateBlock.Block.MarshalRLP(), nil
 }
 
-func (f *fsm) stateTransactions() []*types.Transaction {
-	var txns []*types.Transaction
-
+// applyBridgeCommitmentTx builds state transaction which contains data for bridge commitment registration
+func (f *fsm) applyBridgeCommitmentTx() error {
 	if f.proposerCommitmentToRegister != nil {
-		// add register commitment transaction
-		inputData, err := f.proposerCommitmentToRegister.EncodeAbi()
+		bridgeCommitmentTx, err := f.createBridgeCommitmentTx()
 		if err != nil {
-			f.logger.Error("StateTransactions failed to encode input data for state sync commitment registration", "Error", err)
+			return err
+		}
 
-			return nil
+		if err := f.blockBuilder.WriteTx(bridgeCommitmentTx); err != nil {
+			return fmt.Errorf("failed to apply bridge commitment state transaction. Error: %w", err)
 		}
+	}
 
-		txns = append(txns,
-			createStateTransactionWithData(f.config.StateReceiverAddr, inputData))
+	return nil
+}
+
+// createBridgeCommitmentTx builds bridge commitment registration transaction
+func (f *fsm) createBridgeCommitmentTx() (*types.Transaction, error) {
+	if f.proposerCommitmentToRegister == nil {
```

Comment:
I think this check is redundant, since in the `applyBridgeCommitmentTx` function we check if this is nil.

Reply:
I have added it because of unit tests (just in case somebody calls it without setting `proposerCommitmentToRegister` field) and to prevent any unwanted panics (better to return an error, just in case).

Reply:
I agree with you what you said about panics, but I would not conform the client code itself to tests, especially since we don't  expect this to be called if `proposerCommitmentToRegister` is not set.

Reply:
Fixed [68c80c0](https://github.com/0xPolygon/polygon-edge/pull/1312/commits/68c80c07fcd84bd0d22f00f8bf6000b45befb148)